.github/workflows/ci.yml

```diff
@@ -0,0 +1,68 @@
+name: 'Continuous Integration'
+on: [ push, pull_request ]
+jobs:
+  BuildAndPublish:
+    strategy:
+      matrix:
+        alfresco_version: [ 62, 70, 71, 72, 73 ]
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out
+        uses: actions/checkout@v3
+      - name: Set up JDK
+        uses: actions/setup-java@v3
+        with:
+          java-version: 11
+          distribution: temurin
+      - name: Login to Docker
+        uses: docker/login-action@v2
+        with:
+          registry: private.docker.xenit.eu
+          username: ${{ secrets.CLOUDSMITH_USER }}
+          password: ${{ secrets.CLOUDSMITH_APIKEY }}
+      - name: Build interface  # Execute before integration testing to catch errors early
+        uses: gradle/gradle-build-action@v2.3.0
+        env:
+          XENIT_ARTIFACTORY_USERNAME: ${{ secrets.XENIT_ARTIFACTORY_USERNAME }}
+          XENIT_ARTIFACTORY_PASSWORD: ${{ secrets.XENIT_ARTIFACTORY_PASSWORD }}
+        with:
+          arguments: :apix-interface:build :apix-interface:javadoc
+      - name: Unit test REST API
+        uses: gradle/gradle-build-action@v2.3.0
+        env:
+          XENIT_ARTIFACTORY_USERNAME: ${{ secrets.XENIT_ARTIFACTORY_USERNAME }}
+          XENIT_ARTIFACTORY_PASSWORD: ${{ secrets.XENIT_ARTIFACTORY_PASSWORD }}
+        with:
+          arguments: --info :apix-rest-v1:test
+      - name: Build and test
+        uses: gradle/gradle-build-action@v2.3.0
+        env:
+          XENIT_ARTIFACTORY_USERNAME: ${{ secrets.XENIT_ARTIFACTORY_USERNAME }}
+          XENIT_ARTIFACTORY_PASSWORD: ${{ secrets.XENIT_ARTIFACTORY_PASSWORD }}
+        with:
+          arguments: >-
+            --info
+            :apix-impl:apix-impl-${{ matrix.alfresco_version }}:test
+            :apix-integrationtests:test-${{ matrix.alfresco_version }}:integrationTest
+      - name: Upload Artifact
+        if: always()
```

Comment:
Good question. It was either the choice between: A `success() || failure()` or B `always()` .
A (besides looking confusing) does not upload the report in case of cancellation of the build, which you do when the build hangs, so I went for B.
In the end, it seems it does not matter, because when cancelling or failing integration tests, only the unit test report is uploaded. Still, small preference for B.